### PR TITLE
feat: --no-deprecation

### PR DESCRIPTION
**Pull request by atian25**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change

add `--no-deprecation` as default execArgv to avoid some log like `DeprecationWarning: Buffer() is deprecated` to stderr which will break startup unnecessary.

### CODE REVIEW

File: lib/cmd/start.js [hunk 1]
```diff
@@ -128,10 +128,13 @@ class StartCommand extends Command {
...
     }
 
+    // additional execArgv
+    execArgvObj.deprecation = false; // --no-deprecation
+
     const command = argv.node || 'node';
 
     const options = {
-      execArgv,
+      context.execArgv, // getter for execArgvObj, see https://github.com/node-modules/common-bin/blob/master/lib/command.js#L332
```
Comment (mansonchor) [inline lib/cmd/start.js:137]: 这样写语法没问题？

Comment (atian25) [reply lib/cmd/start.js]: 改了。。。 小重构了下忘记跑了

**Comment by mansonchor** [comment]
CI 跪了

**Review by mansonchor — APPROVED** [review]
LGTM

**Comment by licaide** [comment]
--no-deprecation  运行 pkg 打包eggjs 执行会报错 
如下：
egg-scripts] Starting egg application at /snapshot/pkg-egg3js-example
[egg-scripts] Run node --no-deprecation --trace-warnings /snapshot/pkg-egg3js-example/node_modules/egg-scripts/lib/start-cluster {"no-deprecation":["true",1],"baseDir":"/snapshot/pkg-egg3js-example","framework":"/snapshot/pkg-egg3js-example/node_modules/egg","title":"egg-server-pkg-egg3js-example"} --title=egg-server-pkg-egg3js-example
[egg-scripts] Save log file to /root/logs
[egg-scripts] Wait Start: 1...
[egg-scripts] tail -n 100 /root/logs/master-stderr.log
[egg-scripts] Got error when startup: 
[egg-scripts] node:internal/modules/cjs/loader:949
[egg-scripts]   throw err;
[egg-scripts]   ^
[egg-scripts] 
[egg-scripts] Error: Cannot find module '/home/product/node/test_lcd/--no-deprecation'
[egg-scripts]     at Module._resolveFilename (node:internal/modules/cjs/loader:946:15)
[egg-scripts]     at Function._resolveFilename (pkg/prelude/bootstrap.js:1951:46)
[egg-scripts]     at Module._load (node:internal/modules/cjs/loader:787:27)
[egg-scripts]     at Function.runMain (pkg/prelude/bootstrap.js:1979:12)
[egg-scripts]     at node:internal/main/run_main_module:17:47 {
[egg-scripts]   code: 'MODULE_NOT_FOUND',
[egg-scripts]   requireStack: []
[egg-scripts] }
[egg-scripts] 
[egg-scripts] Node.js v18.5.0
[egg-scripts] 
[egg-scripts] Start got error, see /root/logs/master-stderr.log

**Pull request merged**